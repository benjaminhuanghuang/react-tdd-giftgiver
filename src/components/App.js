import React, { Component } from 'react'
import { Button } from 'react-bootstrap'

export default class App extends Component {
  constructor() {
    super();
    this.state = {
      gifts: []
    };
  }
  addGift = () => {
    const { gifts } = this.state;

    const ids = this.state.gifts.map(gift => gift.id)
    const max_id = ids.legth > 0 ? Math.max(...ids) : 0
    gifts.push({ id: max_id + 1 })

    this.setState({gifts})
  }
  render() {
    return (
      <div>

        <Button className='btn-add' onClick={this.addGift}> Add Gift</Button>
      </div>

    )
  }
}
