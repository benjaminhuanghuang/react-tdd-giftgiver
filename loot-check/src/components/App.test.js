import React from "react"
import { shallow } from 'enzyme'
//
import App from './App'

describe('App', ()=>{
  const app = shallow(<App />)

  it("renders correnctly", () => {
    expect(app).toMatchSnapshot()
  })


  it("constains a Wallet componnet", () => {
    //console.log(app.debug());  // app constains <Connect(Wallet)/>
    expect(app.find('Connect(Wallet)').exists()).toBe(true);
  })
  
})




