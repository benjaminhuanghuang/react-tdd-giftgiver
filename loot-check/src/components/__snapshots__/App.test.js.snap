// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App constains a Wallet componnet 1`] = `ShallowWrapper {}`;

exports[`App renders correnctly 1`] = `ShallowWrapper {}`;
